src/main/java/org/wso2/carbon/securevault/azure/AzureKeyVaultRepository.java

```diff
@@ -0,0 +1,381 @@
+
+            if (log.isDebugEnabled()) {
+                if (StringUtils.isNotEmpty(secretVersion)) {
+                    log.debug("Secret version found. Retrieving the specified version of secret.");
+                } else {
+                    log.debug("Secret version not found. Retrieving latest version of secret.");
+                }
+            }
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Secret version not found. Retrieving latest version of secret.");
+            }
+        }
+
+        return new String[] {secretName, secretVersion};
+    }
+
+    /**
+     * Authenticates to the Key Vault using a credential chain.
+     *
+     * @param properties Configuration properties from file.
+     */
+    public static void authenticateToKeyVault(Properties properties) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Initializing Azure Key Vault connection.");
+        }
+
+        try {
+            readConfigProperties(properties);
+        } catch (UnknownHostException e) {
+            log.error("Error in Key Vault configuration.", e);
+        }
+
+        try {
+            secretClient = new SecretClientBuilder()
+                    .vaultUrl(HTTPS_COLON_DOUBLE_SLASH + keyVaultName + DOT + VAULT + DOT + AZURE + DOT + NET)
+                    .credential(createChosenCredential(credential))
+                    .buildClient();
+        } catch (CredentialUnavailableException e) {
+            log.error("Building secret client failed.", e);
+        }
+
+    }
+
+    /**
+     * Reads Carbon Secure Vault configuration properties.
+     *
+     * @param properties Configuration properties from file.
+     * @throws UnknownHostException if the Key Vault name has not been provided.
+     */
+    private static void readConfigProperties(Properties properties) throws UnknownHostException {
+
+        String legacyProvidersString = properties.getProperty("secretRepositories", null);
+        boolean novelFlag;
+
+        if (StringUtils.isEmpty(legacyProvidersString)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Legacy provider not found. Using novel configurations.");
+            }
+            novelFlag = true;
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Legacy provider found. Using legacy configurations.");
+            }
+            novelFlag = false;
+        }
+
+        String legacyPropertyPrefix = SECRET_REPOSITORIES + DOT + VAULT + DOT + PROPERTIES + DOT;
+        String novelPropertyPrefix = SECRET_PROVIDERS + DOT + VAULT + DOT + REPOSITORIES + DOT +
+                AZURE + PROPERTIES + DOT;
+
+        String propertyCredential = novelFlag ? (novelPropertyPrefix + CREDENTIAL.toLowerCase()) :
+                (legacyPropertyPrefix + CREDENTIAL.toLowerCase());
+        String propertyKeyVaultName = novelFlag ? (novelPropertyPrefix + KEY_VAULT_NAME) :
```

Comment:
Recheck this, not tally with the configs mentioned in READ.ME file.

Reply:
The README.md file only mentions the legacy configurations. This is for the novel configurations.
The README.md has been updated to cover both now.